qiskit/transpiler/passes/optimization/template_matching/forward_match.py

```diff
@@ -404,13 +405,39 @@ def run_forward_match(self):
                 ):
                     continue
 
+                template = None
+
                 # Check if the qubit, clbit configuration are compatible for a match,
                 # also check if the operation are the same.
                 if (
                     self._is_same_q_conf(node_circuit, node_template)
                     and self._is_same_c_conf(node_circuit, node_template)
                     and self._is_same_op(node_circuit, node_template)
                 ):
+                ################################################
+                # Check if parameters match the template or not.
+                # parameter check까지 여기서 해버리자.
+                    temp_match = self.match.copy()
+                    temp_match.append([i,label])
+
+                    substitution = TemplateSubstitution(
+                        [temp_match],
+                        self.circuit_dag_dep,
+                        self.template_dag_dep,
+                        None,
+                    )
+                    
+                ##user_cost_dict = None 
+
+                    template_sublist = [x[0] for x in temp_match]
+                    circuit_sublist = [x[1] for x in temp_match]
+                    circuit_sublist.sort()
```

Comment:
Why is it necessary to sort this list?

Reply:
I thought the shorter have to work first. But I think I wrong, so I erased in the next version.